acasclient/lsthing.py

```diff
@@ -2076,6 +2085,26 @@ def _upload_file_values_from_state_dict(state_dict):
             return state_dict
         self.metadata = _upload_file_values_from_state_dict(self.metadata)
         self.results = _upload_file_values_from_state_dict(self.results)
+    
+    def validate(self, client):
+        """Validate SimpleLsThing dictionary. Currently only checks that all CodeValues that reference ACAS DDicts
+        have known/valid values.
+
+        :param client: Authenticated instance of acasclient.client
+        :type client: acasclient.client
+        """
+        for _, values_dict in self.metadata.items():
```

Comment:
Thanks for the comment, I agree. I've shifted responsibility for fetching the contents of the DDicts up to some SimpleLsThing methods so that this can be done in bulk.

Also, `self.metadata` is the SimpleLsThing dictionary, so it won't have ignored or deleted values. The underlying `_ls_thing` object will, but we're not validating that.